zipkin-collector/rabbitmq/src/test/java/zipkin/collector/rabbitmq/RabbitMqCollectorTest.java

```diff
@@ -0,0 +1,303 @@
+/**
+ * Copyright 2015-2017 The OpenZipkin Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License
+ * is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express
+ * or implied. See the License for the specific language governing permissions and limitations under
+ * the License.
+ */
+package zipkin.collector.rabbitmq;
+
+import com.rabbitmq.client.Channel;
+import com.rabbitmq.client.Connection;
+import com.rabbitmq.client.ConnectionFactory;
+import java.io.IOException;
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.List;
+import java.util.concurrent.CopyOnWriteArraySet;
+import java.util.concurrent.LinkedBlockingQueue;
+import java.util.concurrent.TimeUnit;
+import java.util.concurrent.TimeoutException;
+import java.util.concurrent.atomic.AtomicInteger;
+import org.junit.ClassRule;
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.rules.ExpectedException;
+import org.junit.rules.Timeout;
+import org.testcontainers.containers.GenericContainer;
+import zipkin.Codec;
+import zipkin.Span;
+import zipkin.collector.InMemoryCollectorMetrics;
+import zipkin.collector.rabbitmq.RabbitMqCollector.Builder;
+import zipkin.collector.rabbitmq.RabbitMqCollector.LazyRabbitWorkers.RabbitCollectorStartupException;
+import zipkin.internal.ApplyTimestampAndDuration;
+import zipkin.internal.V2SpanConverter;
+import zipkin.storage.AsyncSpanConsumer;
+import zipkin.storage.AsyncSpanStore;
+import zipkin.storage.SpanStore;
+import zipkin.storage.StorageComponent;
+import zipkin2.codec.SpanBytesEncoder;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static zipkin.TestObjects.LOTS_OF_SPANS;
+import static zipkin.TestObjects.TRACE;
+import static zipkin.collector.rabbitmq.RabbitMqCollector.convertAddresses;
+
+public class RabbitMqCollectorTest {
```

Comment:
there is insulation and also the ability to test locally in helpers like LazyCassandra3Storage might want to do that here